Refactor dismissible notice PMax and adds URLs for PMax links

jorgemd24 opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

This is the follow up of https://github.com/woocommerce/google-listings-and-ads/pull/1492#discussion_r869866530, in that conversation we discussed converting the dismissible notice into a more generic component, this PR is refactoring the PMax notice to make it more reusable so it can be used in other parts of this project. 

This PR also adds the URLs for the links "Learn more about this upgrade" as they were pending to be confirmed by the design team.

See more context in the following links:

- [Design Proposal i2: Performance Max Migration (Milestone 2)](https://woodesigngrowconsultants.wordpress.com/2022/04/20/design-proposal-i2-performance-max-migration-milestone-2/#in-reports)
- [Figma File](https://href.li/?https://www.figma.com/file/eQ9O4m2flzrcAiDMXkOW0m/Google-Listings-%26-Ads-v1.x?node-id=7840%3A127644)
- https://github.com/woocommerce/google-listings-and-ads/pull/1472 & https://github.com/woocommerce/google-listings-and-ads/pull/1492
- https://github.com/woocommerce/google-listings-and-ads/pull/1508

### Detailed test instructions:

#### Links for banners and Inbox notifications

##### Unconverted
1. Check the wp_options table if you have the following option `gla_campaign_convert_status`, otherwise refresh the page.
2. If your status is unconverted: `a:2:{s:6:"status";s:11:"unconverted";s:7:"updated";i:1653037558;}` a banner in the dashboard section should be displayed.
3. Check if the link `Learn more about this upgrade` is redirecting to the right page. `https://support.google.com/google-ads/answer/11576060`

![image](https://user-images.githubusercontent.com/2488994/169703295-877a2ca2-19c1-461c-84cc-983f58419258.png)

4. Go to Tools->Scheduled Actions->search for `wc_gla_cron_daily_notes` in the pending tab.
5. Click run for the pending job: `wc_gla_cron_daily_notes`
6. Go to the Woo Dashboard or your inbox notifications and a new inbox notification should be displayed. Click the link `Learn more about this upgrade` and it should redirect to  `https://support.google.com/google-ads/answer/11576060`

![image](https://user-images.githubusercontent.com/2488994/169394103-8ef9f062-c86c-4585-b949-78391c09fd03.png)

##### Converted

1. Update your status to converted. Update the `gla_campaign_convert_status` option to look similar to `a:2:{s:6:"status";s:9:"converted";s:7:"updated";i:1653037558;}` (the updated timestamp is not needed to be updated)
2. A banner in the dashboard section and in the reports section should be displayed. 

![image](https://user-images.githubusercontent.com/2488994/169703922-3bd84a90-d885-4484-a953-40ebe20e6cfb.png)

![image](https://user-images.githubusercontent.com/2488994/169394318-f98f8493-5d9c-4194-956c-38e3df1ca027.png)

3. Check if the links `Learn more about this upgrade` is redirecting to the right page. `https://support.google.com/google-ads/answer/11576060` in both banners.
4. Go to Tools->Scheduled Actions->search for `wc_gla_cron_daily_notes` in the pending tab.
5. Click in run for the pending job: `wc_gla_cron_daily_notes`
6. Go to the Woo Dashboard or your inbox notifications and a new inbox notification should be displayed. Click the link `Read more about this upgrade` and it should redirect to  `https://support.google.com/google-ads/answer/11576060`

![image](https://user-images.githubusercontent.com/2488994/169703952-a90599eb-0fd0-4791-bd41-4adebc62ec38.png)



### Changelog entry

>Tweak - Refactor dismissible notice

jorgemd24 committed:
Refactor dismissible notice

jorgemd24 committed:
Add test dismissible notice

jorgemd24 committed:
Add read more links migration

jorgemd24 committed:
Update links migration

jorgemd24 committed:
Fix lint js

jorgemd24 committed:
Update defaultDismissedValue

jorgemd24 committed:
Fix typo Dismissible

jorgemd24 committed:
Tweak tests dismissible notice

puntope commented on js/src/components/campaign-conversion-notice/conversion-statuses.js:
```diff
@@ -58,8 +57,7 @@ const AFTER_CONVERSION = {
 	externalLink: {
-		//TODO Update link when it is defined
-		link: '#',
+		link: 'https://support.google.com/google-ads/answer/11576060',
```
💅  This link could be stored as a constant to avoid repetition

jorgemd24 replied:
Added here: [18c8557](https://github.com/woocommerce/google-listings-and-ads/pull/1515/commits/18c8557804b571d5eba6867e440689eab03f1d59)

puntope commented on js/src/components/dismissible-notice/index.js:
```diff
@@ -16,26 +17,30 @@ import localStorage from '.~/utils/localStorage';
+	...rest
 } ) => {
 	const defaultDismissedValue = localStorageKey
-		? !! localStorage.get( localStorageKey )
+		? localStorage.get( localStorageKey ) === 'true'
 		: false;
```
💅 

This could be simplified as 

const defaultDismissedValue = localStorage.get( localStorageKey ) === 'true';

jorgemd24 replied:
Added here: [d30ca34](https://github.com/woocommerce/google-listings-and-ads/pull/1515/commits/d30ca34b4414377e063c99c73d543a4593154101)

puntope reviewed (commented):
Thanks for the changes @jorgemd24 

I could complete the testing for this tho... 

⚠️  Unable to show after migration inbox notification and dashboard banner

- I added `a:2:{s:6:"status";s:11:"unconverted";s:7:"updated";i:1653037558;}` in `gla_campaign_convert_status` 
- I deleted `gla-before-campaign-migration` and `gla-after-campaign-migration` from wc-admin-notices table
- I run scheduled actions `wc_gla_cron_daily_notes`
- I saw the Inbox notice and banner notice
- I clicked on the button to read more
- I didnt' click on the banner read more neither dismiss it
- I refreshed the page and the notice is not there anymore
- I changed  to  `{s:6:"status";s:9:"converted";s:7:"updated";i:1653037558;}` in `gla_campaign_convert_status` 
- I run scheduled actions `wc_gla_cron_daily_notes`
- I didn't see the banner for after  migration ⚠️ neither the inbox, instead I see the before migration notice

jorgemd24 commented:
Hey @puntope, thanks for reviewing this.

>I didn't see the banner for after migration ⚠️ neither the inbox, instead I see the before migration notice

Can you double-check that the status was 'converted' in gla_campaign_convert_status? 

Try the following steps:
1. Update `gla_campaign_convert_status` to {s:6:"status";s:9:"converted";s:7:"updated";i:1653300490;}
2. Refresh the GLA dashboard page
3. Check if `gla_campaign_convert_status`still converted and the banners should be displayed.

I had a similar issue when I was updating manually the `gla_campaign_convert_status` instead of `s:9:"converted"` I set `s:11:"converted"` and it was failing to deserialize therefore it changed to 'unknown' status or 'unconverted' if you have SSC campaigns. 

Let me know your results!

puntope commented:
> Update gla_campaign_convert_status to {s:6:"status";s:9:"converted";s:7:"updated";i:1653300490;}

So finally I saw the problem. I missed `a:2 `part in the string. 

Now the behaviour is correct! 👌 

I left some comments tho regarding code. 💅 

Also was a bit strange that if the string is not correct it shows `BEFORE_CONVERSION` state by default.

jorgemd24 committed:
Add constant read more link

jorgemd24 committed:
Simplify defaultDismissedValue

jorgemd24 commented:
Thanks, @puntope, I added your suggestions!

>Also was a bit strange that if the string is not correct it shows BEFORE_CONVERSION state by default.

Yeah that was because it was failing to deserialize so the status was 'unknown' and this makes to recalculate the conversation status: 

https://github.com/woocommerce/google-listings-and-ads/blob/0777d4079f1fca4217b74ea886a4b28e03512b33/src/API/Google/AdsCampaign.php#L363

I think in your case, probably you have SSC campaigns so that's why it was showing 'unconverted', in my case for example that I removed the old SSC campaigns after 'unknown' it goes to 'converted'.

puntope reviewed (approved):
✅  LGTM 

Thanks for the PR and further adjustments 👌
